lib/redis/connection/ruby.rb

```diff
@@ -286,7 +286,10 @@ def self.connect(host, port, timeout, ssl_params)
           ssl_sock = new(tcp_sock, ctx)
           ssl_sock.hostname = host
           ssl_sock.connect
-          ssl_sock.post_connection_check(host)
+
+          unless ssl_params.has_key?(:verify_hostname) && ssl_params[:verify_hostname] == false
```

Comment:
The last half of the check is benign, and the first part (`ssl_params.has_key?(:verify_hostname)`) _could_ be left out.

Example:

```
irb(main):009:0> ssl_params = { }
{}
irb(main):010:0> ssl_params[:verify_hostname] == false
false #   which is the value we are after
```

Reply:
True, not sure I recall why I checked for the key in the first place, will issue a fixup.

Reply:
See 4bafef2. Note for consistency with the code for that method, a truthy check for `ssl_params` is done I assume because it's expected that `ssl_params` could be null

Reply:
Thanks for putting in this effort!